mri/reconstructors/base.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+# System import
+import warnings
+
+# Package import
+from ..operators.linear.wavelet import WaveletUD2, WaveletN
+from ..optimizers import pogm, condatvu, fista
+from ..optimizers.utils.cost import GenericCost
+
+# Third party import
+from modopt.opt.proximity import SparseThreshold
+from modopt.opt.linear import Identity
+
+
+class ReconstructorBase(object):
+    """ This is the base reconstructor class for reconstruction.
```

Comment:
Do we even have to document this? Why cant we make this an internal thing? After all, all reconstructors derive from here